boot/boot.go

```diff
@@ -42,13 +42,14 @@ const (
 )
 
 // RebootInfo contains information about how to perform a reboot if
-// required
+// required.
 type RebootInfo struct {
-	// RebootRequired is true if we need to reboot after an update
+	// RebootRequired is true if we need to reboot after an update.
 	RebootRequired bool
-	// RebootBootloader will not be nil if the bootloader has something to say on
-	// how to perform the reboot
-	RebootBootloader bootloader.RebootBootloader
+	// BootloaderOptions can be provided if the bootloader has something
+	// to say in how to perform the system reboot, and these are then used
+	// find the correct bootloader.
+	BootloaderOptions *bootloader.Options
```

Comment:
booloader.Options needs to be added json tags

Reply:
I wanted to do this in the reboot PR where this would be actually serialized, it seemed irrelevant for this PR. If you think I should add it here I'll happily do so